### ISSUE TITLE
Segfault when listing files in /proc/self/fd 

### ISSUE THREAD

**Issue by rth** [pyodide/pyodide#200]
Running,
```py
import os
os.listdir('/proc/self/fd')
```
produces the following error `abort({}). Build with -s ASSERTIONS=1 for more info.`, which is effectively a segfault. Running any subsequent command, even `1+1`, will produce the same error message.

After re-compiling with `ASSERTIONS=2`, a more detailed traceback is,
```
E       stackTrace@http://127.0.0.1:33465/pyodide.asm.js:4:22873
E       abort@http://127.0.0.1:33465/pyodide.asm.js:4:11184512
E       ___syscall5@http://127.0.0.1:33465/pyodide.asm.js:4:196773
```
which corresponds to the `open` system call.

It should be noted that, the following works as expected,
```py
>>> os.listdir('/proc/self')
['fd']
```

**Comment by hoodmane** [comment]
This is still an issue:
```
pyodide.asm.js:8 Uncaught abort(TypeError: Cannot read property 'parent' of undefined) at Error
    at jsStackTrace (http://localhost:8001/pyodide.asm.js:8:506153)
    at stackTrace (http://localhost:8001/pyodide.asm.js:8:506360)
    at abort (http://localhost:8001/pyodide.asm.js:8:7435951)
    at ___syscall5 (http://localhost:8001/pyodide.asm.js:8:648188)
    at _open (http://localhost:8001/pyodide.asm.wasm:wasm-function[9730]:0x86a11b)
    at _opendir (http://localhost:8001/pyodide.asm.wasm:wasm-function[9777]:0x86c929)
    at _os_listdir (http://localhost:8001/pyodide.asm.wasm:wasm-function[5721]:0x4078fb)
    at http://localhost:8001/pyodide.asm.wasm:wasm-function[20626]:0x9c3e27
    at _cfunction_vectorcall_FASTCALL_KEYWORDS (http://localhost:8001/pyodide.asm.wasm:wasm-function[1727]:0x18afc9)
    at http://localhost:8001/pyodide.asm.wasm:wasm-function[23488]:0x9cde26
```

**Comment by hoodmane** [comment]
@dalcde Is this an emscripten bug?

**Comment by dalcde** [comment]
I think so. The following program fails to run when compiled with emcc:
```
#include <dirent.h>
#include <sys/types.h>

int main() {
  DIR *dp = opendir("/proc/self/fd");
  return 0;
}

**Comment by dalcde** [comment]
This is https://github.com/emscripten-core/emscripten/issues/10774 . Should we close this?

**Comment by hoodmane** [comment]
Fixed by #1102.
